python/cudf/cudf/core/reshape.py

```diff
@@ -121,7 +128,10 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
             result_columns = result_columns.append(o.columns)
 
         df.columns = result_columns.unique()
-        return df.sort_index()
+        if not ignore_index:
+            return df.sort_index()
+        else:
+            return df
```

Comment:
Without the `sort_index` here we end up with a non-deterministic order so we may need to always sort and then just throw away the index object if `ignore_index` is `True`.